qt/lib/include/verovio-qt/veroviodoc.h

```diff
@@ -0,0 +1,208 @@
+     */
+    int pageCount() const { return m_pageCount; }
+
+    /**
+     * @name Render the given page of this document to the specified deviceContext.
+     */
+    void renderPage(int page, DeviceContext *deviceContext);
+
+    /**
+     * @name Returns if the document has valid data to show.
+     */
+    bool hasValidData() const { return m_hasValidData; }
+
+public slots:
+    /**
+     * @name Setters for public properties.
+     */
+    ///@{
+    void setResourcesDataPath(QString resourcesDataPath);
+    void setFontDirPath(QString fontDirPath);
+    void setFileName(QString fileName);
+    void setFileContent(QString fileContent);
+    void setMusicFont(QString musicFont);
+    void setPageWidth(int pageWidth);
+    void setPageHeight(int pageHeight);
+    void setScale(int scale);
+    void setBorder(int border);
+    void setSpacingStaff(int spacingStaff);
+    void setSpacingSystem(int spacingSystem);
+    void setAdjustPageHeight(bool adjustPageHeight);
+    void setNoLayout(bool noLayout);
+    void setIgnoreLayout(bool ignoreLayout);
+    ///@}
+
+signals:
+    /**
+     * @name This signal is emitted if the document layout changed. VerovioPages attach to this signal to update the
+     * scene graph.
+     */
+    void documentLayoutChanged();
+
+    /**
+     * @name These signals are used internally to notify that the document layout has to be done or the file content has
+     * to be read.
+     */
+    ///@{
+    void documentLayoutInvalidated();
+    void fileContentInvalidated();
+    void fileNameInvalidated();
+    ///@}
+
+    /**
+     * @name Signals for Qt properties.
+     */
+    ///@{
+    void pageCountChanged(int pageCount);
+    void hasValidDataChanged(bool hasValidData);
+    ///@}
+
+protected:
+    // instance of the verovio Toolkit class used to communicate to the C++ library.
+    Toolkit m_verovioToolkit;
+
+private slots:
+    /**
+     * @name Loads or reloads the data stored in the m_content member variable.
+     */
+    void reloadData();
+
+    /**
+     * @name Read the content of the file specified by the fileName property.
+     */
+    void readFile();
+
+    /**
+     * @name Worker function that recalculates the document layout. This function is called asynchronously if
+     * requestDocumentLayout() is called.
+     */
+    void documentRelayout();
+
+private:
+    /**
+     * @name These function trigger an asynchronous read file, reload data or document relayout request.
+     */
+    ///@{
+    void requestReadFile();
+    void requestReloadData();
+    void requestDocumentRelayout();
+    ///@}
+
+    /**
+     * @name Internal setter for the page count.
+     */
+    void setPageCount(int pageCount);
+
+    /**
+     * @name Initialies required fonts.
+     */
+    void initFont();
+
+    /**
+     * @name Setter for hasValidData.
+     */
+    void setHasValidData(bool hasValidData);
+
+private:
+    // configuration parameters for Verovio
+    int m_pageWidth;
```

Comment:
With setOptions passed to the toolkit I think you would not need to have all these members duplicated.

Reply:
That's a good idea. Even without using a JSON string all function calls can be directly delegated to the corresponding function in vrv::Toolkit.
One exception is the pageWidth and pageHeight which actually have a different meaning in verovio-qt, because they take into account the current scale. Thus I renamed them to displayWidth and displayHeight.